Bug 50625 (#450)
ALTIBASE/Documents: ReleaseNotes/eng/Altibase_7_3_0_0_1_Release_Notes.md

|                                                              | Altibase Server | Altibase Client | Software requirements   |
| ------------------------------------------------------------ | :-------------: | :-------------: | :---------------------- |
| **AIX on IBM Power Systems**                                 |                 |                 |                         |
| AIX 6.1                                                      |        ●        |        ●        |                         |
| **Linux x86-64**                                             |                 |                 |                         |
| Red Hat Enterprise Linux 6<br/>Red Hat Enterprise Linux 7<br/>Red Hat Enterprise Linux 8 |        ●        |        ●        | - GNU glibc 2.12 ~ 2.33 |
| **Linux on Power**                                           |                 |                 |                         |
| Red Hat Enterprise Linux 6.5 and higher                      |        ●        |        ●        | - GNU glibc 2.12 ~ 2.33 |
| **Linux on Power** **(Little Endian)**                       |                 |                 |                         |

> | Red Hat Enterprise Linux 7.3 and higher                      |        ●        |        ●        | - GNU glibc 2.17 ~ 2.33 |

Author: glibc 2.12 ~ 2.33 이 맞고, 국문 PDF는 제가 수정해놓겠습니다.

| **HP-UX Itanium (IA-64)**                                    |                 |                 |                         |
| HP-UX 11.31                                                  |        ●        |        ●        |                         |

> Altibase 7.3 server and client both support 64-bit only.
>
> Altibase 7.3 is compatible with minor versions of Red Hat Enterprise Linux 6, 7 and 8.
>
> Java 버전: Altibase 7.3 is compatiable with JDK 1.8 and higher.

<br/><br/>

## 2. Release Nots

### 2.1 New Features

> #### 2.1.1 AKU(Altibase Kubernetes Utility)

Author: aku 버전은 Altibase 버전과 동일하게 되어 있어요.

> AKU(Altibase Kubernetes Utility) is a utility that helps with synchronizing data or resetting synchronization information when Pods start and terminate in a Statefulset in Kubernetes.

Reviewer: start and terminate를 start or terminate로만 고쳤습니다. 이 코멘트는 리졸브 하지 않을테니 다시 한번만 확인해주세요(initialization/resetting 선택+during scaling이라는 정보 포함 여부)

#### 2.1.2 AltiShapeLoader 1.0

> altiShapeLoader is developed for importing and exporting shapefiles<sup id="shapefile1">[[1]](#shapefile)</sup> and is based on the GeoTools, which is a Java-based open source framework.

Author: chatGPT 이용했었는데, 검토가 미흡했네요. altishapeloader 영문매뉴얼의 overview 내용을 적는게 좋을것 같아서, 아래의 문장으로 하겠습니다.
altiShapeLoader is a tool used to import and export shapefile[1](https://github.com/ALTIBASE/Documents/blob/master/Manuals/Tools/Altibase_release/eng/altiShapeLoader%20User's%20Manual.md#shapefile), which is programmed based on GeoTools, open-source library based on Java.

> altiShapeLoader is developed for importing and exporting shapefiles<sup id="shapefile1">[[1]](#shapefile)</sup> and is based on the GeoTools, which is a Java-based open source framework.

Author: 소윤님.
altiShapeLoader is a tool used to import and export shapefile<sup id="shapefile1">[[1]](#shapefile)</sup>, which is programmed based on GeoTools, open-source library based on Java.  로 수정해주세요~

> altiShapeLoader is developed for importing and exporting shapefiles<sup id="shapefile1">[[1]](#shapefile)</sup> and is based on the GeoTools, which is a Java-based open source framework.

Reviewer: 수정했습니다~

#### 2.1.4 OpensSSL 3.0.8 Support

To enhance security, Altibase 7.3 supports the latest version of OpenSSL 3.0.8, and no longer provide support for OpenSSL versions 1.0.x. Altibase now extends its protocol support to include TLS 1.3, in addition to TLS 1.0 and 1.2. If users want to specify particular cipher algorithms for TLS 1.3, please set them in the server property, SSL_CIPHER_SUITES. For more detailed information, please refer to [**Altibase SSL TLS User's Guide** - server properties to connect over ssl](https://github.com/ALTIBASE/Documents/blob/master/Manuals/Altibase_7.3/eng/Altibase%20SSL%20TLS%20User's%20Guide.md#step-2-set-server-properties-to-connect-over-ssl).

> Additionally, Altibase supports the FIPS(Federal Information Processing Standards) module. To use the FIPS module with SSL, you must set the SSL_LOAD_CONFIG property to 1. Refer to [Altibase SSL TLS User's Guide - Step4. FIPS module with SSL]().

Author: user가 맞겠네요~

> Additionally, Altibase supports the FIPS(Federal Information Processing Standards) module. To use the FIPS module with SSL, you must set the SSL_LOAD_CONFIG property to 1. Refer to [Altibase SSL TLS User's Guide - Step4. FIPS module with SSL]().

Reviewer: 타사 매뉴얼 등을 살펴본 결과 user보다 users가 더 많이 쓰이는 것 같아서 우선 users로 통일하여 수정했습니다~

</br>

#### 2.1.5 Funtionality Improvement - SQL Extension<b id="215"></b>

##### VARRAY TYPE

Within stored procedures, the VARRAY type is now supported as a new user-defined type. The VARRAY type is capable of storing a sequence of data with the same data type. For more details, please refer to [Stored Procedures Manual - VARRAY]().

##### Anonymous Block 

> An anonymous block is a form of a stored procedure composed of a body block without a header, declared with a structure like DECLARE...BEGIN...END;. Anonymous blocks do not create PSM (Persistent Stored Module) objects, are not stored in the database, and do not return a value in the RETURN clause. Unlike stored procedures, anonymous blocks allow the use of INPUT, OUTPUT, and INOUT bind variables.

Author: 네, 단순 anonymous block에 대한 설명을 기술한 내용입니다. 수정하지 않아도 될것 같네요~

##### Internal Mode in C/C++ External Procedure

> In Internal mode within External Procedures, it directly loads dynamic libraries and invokes external procedures from Altibase Server without the need for an agent process, resulting in improved efficiency compared to external mode. For more details, refer to [**External Procedures Manual**](https://github.com/ALTIBASE/Documents/blob/master/Manuals/Altibase_7.3/eng/External%20Procedures%20Manual.md).

Reviewer: In Internal mode within External Procedures, it directly loads dynamic libraries and invokes external procedures from Altibase Server without the need for an agent process, resulting in improved efficiency compared to external mode. 
->
The internal mode procedure operates faster compared to the external mode by directly loading dynamic libraries and invoking external procedures from the Altibase server without agent processes, resulting in improved efficiency. 

이 문장으로 최종 수정했습니다. 검수 부탁드립니다.

##### Multiple Delete, Update

Provides support for multiple delete, multiple update statements. Refer to [**SQL Reference Manual** - multiple_delete](https://github.com/ALTIBASE/Documents/blob/master/Manuals/Altibase_7.3/eng/SQL%20Reference.md#multiple_delete) , [multiple_update](https://github.com/ALTIBASE/Documents/blob/master/Manuals/Altibase_7.3/eng/SQL%20Reference.md#multiple_update).

##### Regular Expression for Korean searching

Provides PCRE2 compatibility mode for support Regular Expressions used in Korean searching. Refer to [**SQL Reference Manual**](https://github.com/ALTIBASE/Documents/blob/master/Manuals/Altibase_7.3/eng/SQL%20Reference.md).

##### Fetch Across Rollback

To address the 'Fetch out of sequence' error issue, we now provide the 'fetch across rollback' feature.

##### Improved Queue Functionality with Delete Statement Control

> Queue functionality has been enhanced to allow the use of DELETE statements. Additionally, new 'DELETE ON' and 'DELETE OFF' clauses have been introduced to control the execution of DELETE statements within queues. Refer to [**SQL Reference Manual**](https://github.com/ALTIBASE/Documents/blob/master/Manuals/Altibase_7.3/eng/SQL%20Reference.md) and V$QUEUE_DELETE_OFF performance view for more information.

Reviewer 2: Refer to에 PCRE2는 Please가 있고, 여기는 없네요
하나로 통일하는것이 좋겠습니다

##### Sequence Restart Statement

Supports the restart_clause with ALTER SEQUENCE statement. Refer to [**SQL Reference Manual** - restart_clause](https://github.com/ALTIBASE/Documents/blob/master/Manuals/Altibase_7.3/eng/SQL%20Reference.md#restart_clause).

</br>

#### 2.1.6 Funtionality Improvement - Spatial SQL<b id="216"></b>

##### SRID(Spatial Reference Identifier) interface

> SRID (Spatial Reference Identifier) is an identifier used to distinguish spatial objects, represented by a 4-byte integer, and can be applied to GEOMETRY columns. You can specify the SRID when creating a table and also alter the SRID using the ALTER TABLE statement.

Author: 4-byte range , user 통합

> SRID (Spatial Reference Identifier) is an identifier used to distinguish spatial objects, represented by a 4-byte integer, and can be applied to GEOMETRY columns. You can specify the SRID when creating a table and also alter the SRID using the ALTER TABLE statement.

Reviewer: 4-byte range로 수정하고 you는 users로 수정했습니다.

The support for SRID introduces new ways to represent GEOMETRY data types:

- Extended Well-Known Text (EWKT) format: It includes SRID (Spatial Reference Identifier) information in the WKT format to represent spatial objects.
- Extended Well-Known Binary (EWKB) format: It includes SRID (Spatial Reference Identifier) information in the WKB format to represent spatial objects.

##### Spatial Functions

New functions have been introduced as follows.

* ASEWKT
* ASEWKB
* GEOMFROMEWKT
* GEOMFROMEWKB
* SETSRID
* SRID
* ST_Collect
* ST_IsCollection
* ST_LinestringFromWKB
* ST_MakeEnvelope
* ST_MakeLine
* ST_MakePoint
* ST_Point
* ST_PolygonFromText
* ST_Transform

##### Spatial object Creation Functions

New functions have been introduced as follows.

* ACSGetGeometrySRID

#### 2.1.7 Funtionality Improvement - Replication<b id="217"></b>

##### DDM Synchronization

> DDL Synchronization is now possible through replication. To use this feature, you must set the REPLICATION_DDL_SYNC property to 1 on each node. Also, the REPLICATION_DDL_ENABLE property of each node must be set to 1 and the REPLICATION_DDL_ENABLE_LEVEL property must be set to the same.

Reviewer: 수정했습니다.

To use DDL Synchronization, the following constraints must be verified:

- Replication must be operational on the nodes where DDL Synchronization is performed.
- The table names on both local and remote nodes for DDL Synchronization must match.
- The table partition names on both local and remote nodes for DDL Synchronization must match.

> - The username for replication target users must be the same for DDL Synchronization.

Author: DDL 복제를 수행할 이중화 대상 사용자의 이름이 같아야 한다. 국문도 이상하네요.. "DDL 복제를 수행할 테이블의 소유자 이름이 같아야 한다."  DDL 복제를 수행할 이중화 대상 테이블의 소유자 이름을 'user name' 이라 하는데,  owner name 이 더 맞을 것 같습니다. replication creation 문법에서 user_name으로 명칭하고 있긴합니다. 
->The owner names of tables on both local and remote nodes for DDL Synchronization must match.

> - The username for replication target users must be the same for DDL Synchronization.

Reviewer: 수정했습니다. 리졸브하겠습니다.

- Only one node at a time can perform DDL Synchronization.
- The values of the REPLICATION_DDL_ENABLE and REPLICATION_DDL_ENABLE_LEVEL properties must be identical for each node.

> - The Altibase Server version (5 digits) must be the same.

Author: 국문을 고쳐야 합니다. 서버 버전이 동일해야 합니다. must be same 이나 must be identical. 이 낫겠네요.

- DDL Synchronization is not permitted when using the Propagation option.

#####  RECEIVE_ONLY option

> A new option, RECEIVE_ONLY, has been introduced to prevent the transmission of transaction logs for changing data to other nodes. Creating a replication with the RECEIVE_ONLY option means it does not read transaction logs, ensuring that it does not affect the other node, even in the case of issues such as network failures. 

Reviewer 2: 트랜잭션 로그가 시스템에 영향을 미치지 않는게 아니라 RECEIVE_ONLY로 이중화를 생성하는 것이 시스템에 영향을 미치지 않는다가 맞는것 같습니다.

국문에도 그렇게 나와 있습니다.
설명
이중화를 수신 전용 옵션으로 설정하여 다른 노드로 변경 데이터를 전송하지 않도록 한다.

수신 전용으로 이중화를 생성하면 로그를 읽지 않으므로 네트워크 장애 등의 이중화 이슈가 발생하여도 시스템에 영향을 주지 않는다.

원문의 ensuring 다음의 that절의 it도 creating a replication with RECEIVE_ONLY를 가리키는것 같습니다.

확인 바랍니다.

> A new option, RECEIVE_ONLY, has been introduced to prevent the transmission of transaction logs for changing data to other nodes. Creating a replication with the RECEIVE_ONLY option means it does not read transaction logs, ensuring that it does not affect the other node, even in the case of issues such as network failures. 

Reviewer: If users create a replication with the RECEIVE_ONLY option on the remote server, the local server does not read the transaction logs of the remote server. Therefore, any replication issues such as network failures do not affect the local server system.

해당 문장 이렇게 바꿨습니다!

#### 2.1.8 Application Development Interface Extension and Improvements<b id="218"></b>

##### InfiniBand Support

Supports Infiniband, which is based on Remote Direct Memory Access (RDMA) communication for high-speed data communication.

##### New JDBC Driver

- **Auto-loading of JDBC driver class**

  Automatic driver loading using the META-INF/services/java.sql.Driver file , eliminating the need for explicit Class.forName() class loading.

- **Wrapper Pattern Support**

>   Supports the JDBC 4.0 standard interface for obtaining references to implementation objects from proxies. This allows obtaining JDBC objects from proxy objects created in Connection Pool.

Reviewer: 수정했습니다. 리졸브 하겠습니다~

  ```java
  try (Connection sWrappedCon = dbPool.getConnection()) {
      if (sWrappedCon.isWrapperFor(AltibaseConnection.class)) {
          AltibaseConnection connection = sWrappedCon.unwrap(AltibaseConnection.class);
          ...
          ...
  }
  ```

- **National Character Set Support**

  Supports for the standard multilingual processing interface in the JDBC 4.0 specification.

- **Aborting Connections**

  Supports for the Connection.abort() interface for asynchronously terminating the physical connection to the database.

- **Standard Socket Network Timeout API Support**

  Support for the standard interface Connection.setNetworkTimeout() to set the socket response wait time from the database server.

- **Connection Management Enhancements**

  Support for Connection.isValid() to perform validation on Connection objects without a Validation Query

- **Large Update Counts Support**

  Support for executeLargeUpdate() and executeLargeBatch() for updating large numbers of records

- **Set Client Information Support**

  Support for configuring client application attributes (name) using Connection.setClientInfo()

> - **java.sql.SQLType interface Support**

Reviewer: Altibase 7.3 supports AltibaseJDBCType, which implements the JDBC 4.2 standard interface java.sql.SQLType. 
으로 수정했습니다.

- **Automatic JDBC resource release using the Try-with-resources statement**

  ```java
  try (Statement stmt = con.createStatement()) {
        ResultSet rs = stmt.executeQuery(query);
        while (rs.next()) {
          String coffeeName = rs.getString("aaa");
          int supplierID = rs.getInt("bbb");
        }
      }
  }
  ```

> - **Support for using an enhanced for-each loop with SQLException**

Reviewer: Enhanced For-Each Loop With SQLException 로 최종 수정했습니다. 검수 부탁드립니다.

  ```java
  catch(SQLException ex) {
       for(Throwable e : ex ) {
          LOG.error("Error occurred: " + e);
       }
  }
  ```

</br>

#### 2.1.9 Functionality Improvement - Stored Procedures<b id="219"></b>

##### DBMS_STANDARD package

Offers a function for checking trigger events.

##### DBMS_METADATA package

A function for export object creation DDL statements and GRANT statements from database dictionary.

##### DBMS_SQL_PLAN_CACHE package

Provides a stored procedure that functions to keep or delete a specific Execution Plan in the SQL Plan Cache.

##### print_enable/print_disable procedure in DBMS_OUTPUT pakcage

> Added the 'print_enable' and 'print_disable' procedures to enable or disable the 'println' function within PSM. These procedures are executed on a per-session basis.

Author: 제목: Add print_enable/print_disable procedure in DBMS_OUTPUT Package
내용: DBMS_OUTPUT package now includes print_enable/print_disable procedures which offer the functionality to enable or disable the 'println' function within the PSM. These procedures are executed on a per-session basis

> Added the 'print_enable' and 'print_disable' procedures to enable or disable the 'println' function within PSM. These procedures are executed on a per-session basis.

Reviewer: 수정하고 목차 업데이트 했습니다.

> Added the 'print_enable' and 'print_disable' procedures to enable or disable the 'println' function within PSM. These procedures are executed on a per-session basis.

Reviewer: Add ~ in 보다 Add ~ to가 문맥상 더 맞아서 to로 수정했습니다.

##### sleep2 procedure in the DBMS_LOCK package

> Added system stored procedure 'sleep2' to support microsecond sleep.

Author: #### Add sleep2 procedure in the DBMS_LOCK package

DBMS_LOCK package now includes sleep2 procedure to support microsecond sleep.

> Added system stored procedure 'sleep2' to support microsecond sleep.

Reviewer: 수정하고 목차 업데이트 했습니다.

> Added system stored procedure 'sleep2' to support microsecond sleep.

Reviewer: Add ~ in 보다 Add ~ to가 문맥상 더 맞아서 to로 수정했습니다.

- Enhanced In-Place MVCC Operation for Disk Database Alterations
- Elimination of Unnecessary Transaction Logging during INSERT/UPDATE Transactions
- Improved Memory Allocation/Deallocation Bottlenecks during Transaction Log File Compression
- Enhanced Commit and Garbage Collection Thread Bottlenecks
- Improved Transaction Performance on Memory Databases
  - Eliminated bottlenecks in functions that trigger disk reads
  - Group Commit Log

##### Index Performance Improvement

* Reduced the time and memory usage for POINTER BASE index creation during server startup.
* Reduced the time and memory usage for VALUE BASE index creation during server startup.

##### Database Startup

Improved the management of threads used for index building during server startup.

##### Improve Transaction Performance for Volatile and Non-Volatile Memory Table

Improved volatile/non-volatile memory table transaction performance by simplifying the memory table object identifier tracking steps.

##### DEQUEUE in Parallel

eliminated bottlenecks that occur during parallel DEQUEUE operations.

##### Reduction in Prepare Time for Common Subexpression Elimination

> CSE (Common Subexpression Elimination) is an optimization feature that identifies and removes redundant conditional expressions in query conditions. Improved the CSE execution algorithm and enhance the performance of related queries.

Reviewer: @jina-altibase 
[https://github.com/ALTIBASE/Documents/pull/450/commits/f4289bac890e402771221fe27a358b5fee2c7179](url)

위의 커밋로그에 관련 문장들을 1차적으로 수정했습니다. 중간에 OLTP Scalability~ DEQUEUE in Parallel 구간은 국문 자체도 ~를 개선 형식으로 짧게 쓰여져 있어 굳이 주어를 쓰지 않고 수정해보았습니다. 한국어로 번역하면 향상된 무엇무엇, 감소된 시간과 사용량.. 이런 느낌인데 1차적인 검수를 부탁드립니다.

- INDEX_INITTRANS maximum value has been increased from 30 to 50.
- Default and maximum values for INDEX_MAXTRANS have been changed from 30 to 50.

#### 2.1.14 DBeaver Package

Provide a package for DBeaver on Windows.

</br>

### 2.2 Changes

The following describes the features that DBAs and developers need to be aware of, which include additions, modifications, and removals.

#### 2.2.1 Database Version

Version by Database Component

| Altibase server/client Version | Database Binary Version | Meta Version | Communication Protocol Version | Replication Protocol Version |
| :----------------------------: | :---------------------: | :----------: | :----------------------------: | :--------------------------: |
|           7.1.0.8.8            |          6.5.1          |    8.11.1    |             7.1.7              |            7.4.7             |
|           7.3.0.0.1            |          7.3.0          |    9.3.1     |             7.1.8              |            7.4.9             |

#### 2.2.2 Compatibility

##### Database Binary Version

The database binary version indciates the compatibility of the database image file and log file.

> The database binary version has been updated due to enhancements in the logging structure of log files. As a result, Altibase 7.3 and earlier versions are not compatible, requiring migration efforts when upgrading to Altibase 7.3.

Author: 사용자가 직접 마이그레이션 할 수가 없을거예요. migration 작업이 꽤 크고, 기술지원으로요청하기때문에 require migration efforts 로 표현했습니다.
logging structure가 수정되었기 때문에,  호환되지 않고, 업그레이드 하려면 migration이 필요하다.는 표현입니다.
Altibase 7.3 and earlier versions are not compatible, so migration is required to upgrade from the earlier version. 는 어떠한지요?

> The database binary version has been updated due to enhancements in the logging structure of log files. As a result, Altibase 7.3 and earlier versions are not compatible, requiring migration efforts when upgrading to Altibase 7.3.

Reviewer: The database binary version indicates the compatibility between database image files and log files. The updated database binary version of Altibase 7.3 is not compatible with the lower version anymore because the logging structure of log files is improved. Therefore, if users want to upgrade the existing Altibase version to Altibase 7.3, users should request migration tasks from us(혹은 기술지원팀) in advance.

우선 460번째 줄의 문장을 명확하게 수정했습니다. 뒤의 문장은 설명해주신 내용을 바탕으로 다시 구성해봤는데 한번 확인해주세요~

> The database binary version has been updated due to enhancements in the logging structure of log files. As a result, Altibase 7.3 and earlier versions are not compatible, requiring migration efforts when upgrading to Altibase 7.3.

Author: improved -> changed. 가 더 나을것 같습니다.
음,, 마이그레이션을 반드시 기술지원 거쳐서 해야 한다고 명시할필요는 없지 않을까 해요.. (사실은 꼭 거치지만요,,)
그래서 Therefore, if users want to upgrade the existing Altibase version to Altibase 7.3, migration tasks are required.마이그레이션이 필요 하다. 정도로만 작성하는게 좋을 것 같습니다.

> The database binary version has been updated due to enhancements in the logging structure of log files. As a result, Altibase 7.3 and earlier versions are not compatible, requiring migration efforts when upgrading to Altibase 7.3.

Reviewer: 넵 반영해서 수정했습니다.

##### Meta Version

Given that the major version of Meta has changed, it is necessary to reconfigure the metadata when upgrading from the earlier version to Altibase 7.3.

##### Communication Protocol Version

> The patch version of the communication protocol version has been changed. The same major and minor versions ensure client backward compatibility.

Reviewer: only if를 사용하여 수정하였습니다.

> Client backward compatibility ensures that user applications developed using a older version of the Altibase library work well when connected to a newer version of Altibase.

##### Replication Protocol Version

Major and minor version of replication protocol version has not been changed, but the patch version has been updated. Thus, LAZY mode replication is compatible between older and newer versions, but EAGER mode replication and other additional features that require the same replication protocol version are not compatible. 

> ###### Altibase Replication Backward Compatibility
>
> Only LAZY mode replication guarantees backward compatibility.
>
> While EAGER mode replication, DDL synchronization and Offline Replication do not ensure backward compatibility since they require the same replication protocol version.

> 

Author: DDL synchronization does not ensure the backward compatibility because it requires all three 자리(digits?) of the replication protocol version to match.
-> DDL synchronization does not ensure the backward compatibility because it requires the identical replication protocol version. 으로 해도 되지 않을까요? 아니면 it requires that all of replication protocol versions are same.

Additional features -> Replication additional feature 가 더 나을것 같습니다.. 
 그런데,, 일단 이부분은,,

##### Replication Protocol Version
Replication protocol version indicates the compatibility of replication for both backward compatibility and compatibility with additional replication features.

Altibase replication in Lazy mode ensure the Altibase replication backward compatibility, because there is no change of major and minor version of replication protocol. However, it does not ensure the compatibility with replication additional features including Eager mode replication, because patch version has been changed.

이것만 작성하고, "Altibase Replication Backward Compatibility"는 추후에 다시 보강할께요. 국문도 좀 다듬어야 할거 같아서, 영문에서는 뺐었습니다.

> 

Reviewer: "Altibase Replication Backward Compatibility"를 제외한 Replication Protocol Version 본문을 수정하였습니다.

- [lob_null_select](https://github.com/ALTIBASE/Documents/blob/master/Manuals/Altibase_7.3/eng/JDBC%20User's%20Manual.md#lob_null_select)
  - When a LOB column's value is NULL, the JDBC connection property 'lob_null_select' has been introduced to control the behavior of getBlob() and getClob() functions. 
  - In Altibase 7.3, the default value is 'off', which means getBlob() and getClob() functions return 'NULL'.
  - In Altibase 7.1, the default value is 'on', which means thease functions return LOB objects.

- [getprocedures_return_functions](https://github.com/ALTIBASE/Documents/blob/master/Manuals/Altibase_7.3/eng/JDBC%20User's%20Manual.md#getprocedures_return_functions)
  - Configuration for including function information in the results of DatabaseMetaData.getProcedures() and getProcedureColumns() is provided. While the JDBC API Specification 4.2 standard excludes function information, Altibase 7.3 JDBC Driver includes it for client backward compatibility. To exclude function information in accordance with the JDBC 4.2 standard, set this property value to false.

###### CLIENT_TYPE

> The CLIENT_TYPE of an Altibase 7.3 JDBC session is NEW_JDBC42.

Reviewer: - 빠진 문장

Altibase 7.3 JDBC Driver 를 이용하여 컴파일 또는 실행한 경우 V$SESSION의 CLIENT_TYPE 값은 NEW_JDBC42 로 조회해야 한다.
내용 누락
꼭 필요한 내용은 아닌 것 같아서 점검의 의미로 기록했습니다.

</br>

##### Changes in SQL Results and Execution Plans

- Improved SQL Performance When Using the ORDER BY Clause in Subquery Inline Views

  This change has an impact on the execution plan of SQLs that utilize it, with the SORT plan node being eliminated within the SUBQUERY FILTER.

- Optimization of Nested LEFT OUTER JOIN Operations

  This change can result in alterations to the execution plan and potential differences in the SQL execution results for SQLs affected by it.

- Modifications and Additions to Subquery Unnesting Functionality

  This change can lead to alterations in the execution plans of SQLs influenced by it.

</br>

##### Replication Restrictions

###### Replication Restrictions between Altibase 7.1 and Altibase 7.3

Because the replication protocol version has been changed, DDL synchronization are not supported between Altibase 7.1 and Altibase 7.3.

> Because the database binary version has been changed, offline replication are not supported between Altibase 7.1 and Altibase 7.3.

Reviewer: - 빠진 문장

두 문장 다 이중화 프로토콜 버전 세 자리가 일치해야 한다는 내용이 없습니다.. 점검의 의미로 기록합니다.

Reviewer 2: 1. Because the replication protocol version has been changed,
Because the database binary version has been changed
위 두문장을 빼도 되지 않을까요? 위에 호환성 부분에 내용이 나와 있는데, 아래와 같이 한 문장으로 써도 될것 같습니다.
DDL synchronization and offline replication is not supported between Altibase 7.1 and Altibase 7.3.
2. 호환성 관련 문장에서는 EAGER일때 DDL sync와 Offline Replication이 호환이 안된다고 나와 있는데 제약 사항에는 그런말이 없어서 LAZY일때는 DDL sync와 Offline Replication이 호환이 된다는 건지 헷갈립니다. 
3. 호환성에는 Offline Replication, 제약에는 offline replication으로 대문자 사용이 다르게 되어 있습니다.

Reviewer: 1. 아예 이유를 빼도 될 것 같습니다. 아니면 Replication Restriction 절의 위치를 2.2.2 Compatibility으로 옮기는 건 어떨까요? 내용 상 2.2.2 절이 더 관련성 있어 보입니다. Altibase Replication Backward Compatibility를 아예 헤딩5 수준의 절로 올리고 이 내용을 포함시키는 것도 좋을 것 같습니다. 
2.  추후 수정 시 DDL synchronization and offline replication are not supported between ~ when the running replication mode is EAGER. 로 수정하겠습니다.
3. 추후 수정시 통일하여 수정하겠습니다. 저는 offline replication으로 통일하고 싶은데 이견 있으시면 코멘트 부탁드립니다.

Reviewer 2: offline replication이 다른 db에서도 쓰이는 일반적인 말이면 소문자로 아니면 대문자로 하는게 좋을것 같네요

Reviewer 2: 구글링해보니 Offline Replication으로 많이 쓰는것 같네요
Altibase의 하나의 기능으로써의 명칭을 지칭하는 것이므로 Offline Replication이라고 쓰는게 더 낫다는 생각입니다.

Author: 다시 읽어봤는데.. Replication Restriction 절의 위치를 2.2.2 Compatibility으로 옮기는 건 어떨까요?  -> 이건 아니것 같고요.. SRID가 껴 있어서요,,
그래서 실장님 의견처럼 because ~~ 는 빼고, 지원안되는것만 쓰는게 좋을 것 같네요. 아래와 같이요~

###### Replication Restrictions Between Altibase 7.1 and Altibase 7.3
DDL synchronization and Offline replication is not supported between Altibase 7.1 and Altibase 7.3.

Reviewer: 코멘트의 내용을 반영하여 위치 변경 없이
Replication Restrictions Between Altibase 7.1 and Altibase 7.3
DDL synchronization and Offline Replication are not supported between Altibase 7.1 and Altibase 7.3 when the running replication mode is EAGER.
로 수정했습니다. 이견 없으시면 이 코멘트는 리졸브하겠습니다.

Author: when the running replication mode is EAGER. 이건 아니예요..

정확히는 EAGER mode replication, DDL synchronization and Offline Replication are not supported between Altibase 7.1 and Altibase 7.3. 입니다.

Author: 제가 suggestion 으로 코멘트 달았으니깐, 이 리뷰는 resolve해도 될것 같아요~

###### Replication Restrictions between Altibase 6.5.1 and Altibase 7.3

> Replication from Altibase 7.3 to Altibase 6.5.1 may fail when the target table contains data with SRID values.

Reviewer: - 빠진 문장

Altibase 이중화 하위 호환성 보장에 따라 Altibase 6.5.1와  Altibase 7.3 간 단방향 및 양방향 LAZY 모드 이중화는 가능하다.
이 내용이 누락됐습니다. 점검의 의미로 기록합니다.

According to the backward compatibility assurance of Altibase replication, both onw-way and two-ways replication between Altibase 6.5.1 and Altibase 7.3 in LAZY mode are possible.

Author: restriction이라서 지원하지 않는것만 기록했습니다. 국문도 지원하지않는것만 적는것이 맞을 것 같습니다. (왜냐면, 이중화 호환성에도 한번더 언급되어 있기때문입니다.)

-   [V$REPL_REMOTE_META_COLUMNS](https://github.com/ALTIBASE/Documents/blob/master/Manuals/Altibase_7.3/eng/General_Reference-2.The%20Data%20Dictionary.md#vrepl_remote_meta_columns)
-   [V$REPL_REMOTE_META_INDEX_COLUMNS](https://github.com/ALTIBASE/Documents/blob/master/Manuals/Altibase_7.3/eng/General_Reference-2.The%20Data%20Dictionary.md#vrepl_remote_meta_index_columns)
-   [V$REPL_REMOTE_META_INDICES](https://github.com/ALTIBASE/Documents/blob/master/Manuals/Altibase_7.3/eng/General_Reference-2.The%20Data%20Dictionary.md#vrepl_remote_meta_indices)
-   [V$REPL_REMOTE_META_ITEMS](https://github.com/ALTIBASE/Documents/blob/master/Manuals/Altibase_7.3/eng/General_Reference-2.The%20Data%20Dictionary.md#vrepl_remote_meta_indices)
-   [V$REPL_REMOTE_META_REPLICATIONS](https://github.com/ALTIBASE/Documents/blob/master/Manuals/Altibase_7.3/eng/General_Reference-2.The%20Data%20Dictionary.md#vrepl_remote_meta_replications)

##### Removed Performance Views

* V$ST_ANGULAR_UNIT
* V$ST_AREA_UNIT
* V$ST_LINEAR_UNIT

</br>

### 2.3 Packages

> | OS    | CPU                       | File Names                                                  |

Author: 서버/클라이언트 열 추가해서 표 수정 반영해주세요~

##### Internal Mode in C/C++ External Procedure

The internal mode procedure operates faster compared to the external mode by directly loading dynamic libraries and invoking external procedures from the Altibase server without agent processes, resulting in improved efficiency. For more details, refer to [**External Procedures Manual**](https://github.com/ALTIBASE/Documents/blob/master/Manuals/Altibase_7.3/eng/External%20Procedures%20Manual.md).

##### Multiple Delete, Update

Provides support for multiple delete and multiple update statements. Refer to [**SQL Reference Manual** - multiple_delete](https://github.com/ALTIBASE/Documents/blob/master/Manuals/Altibase_7.3/eng/SQL%20Reference.md#multiple_delete), [multiple_update](https://github.com/ALTIBASE/Documents/blob/master/Manuals/Altibase_7.3/eng/SQL%20Reference.md#multiple_update).

##### Regular Expression for Korean Searching

> To support searching a regular expression in Korean,  Altibase provides PCRE2 compatibility mode. The PCRE2 compatibility mode supports the regular expression syntax of the PCRE2 library. Please refer to [**SQL Reference Manual**](https://github.com/ALTIBASE/Documents/blob/master/Manuals/Altibase_7.3/eng/SQL%20Reference.md) for more information.

Author: to support searching Korean strings with regular expressions
```suggestion
To support searching Korean strings with regular expressions,  Altibase provides PCRE2 compatibility mode. The PCRE2 compatibility mode supports the regular expression syntax of the PCRE2 library. Please refer to [**SQL Reference Manual**](https://github.com/ALTIBASE/Documents/blob/master/Manuals/Altibase_7.3/eng/SQL%20Reference.md) for more information.
```

A new function has been introduced as follows.

* ACSGetGeometrySRID

#### 2.1.7 Functionality Improvement - Replication

##### DDL Synchronization

DDL Synchronization is now possible through replication. To use this feature, users must set the REPLICATION_DDL_SYNC and REPLICATION_DDL_ENABLE properties on each node to 1. Also, the REPLICATION_DDL_ENABLE_LEVEL property of each node must be set the same.

To use DDL Synchronization, the following constraints must be verified:

- Replication must be operational on the nodes where DDL Synchronization is performed.
- The table names on both local and remote nodes for DDL Synchronization must match.
- The table partition names on both local and remote nodes for DDL Synchronization must match.
- The owner names of tables on both local and remote nodes for DDL Synchronization must match.
- Only one node can perform DDL Synchronization at a time.
- The values of the REPLICATION_DDL_ENABLE and REPLICATION_DDL_ENABLE_LEVEL properties must be identical for each node.
- The Altibase Server version(5 digits) must be the same.
- DDL Synchronization is not permitted when using the Propagation option.

#####  RECEIVE_ONLY Option

> A new option, RECEIVE_ONLY, has been introduced to prevent the transmission of transaction logs for changing data to other nodes. When creating replication with the RECEIVE_ONLY option, transaction logs do not affect other nodes because they are not read. Therefore, any replication issues such as network failures do not affect the system.

Reviewer: ```suggestion
A new option, RECEIVE_ONLY, has been introduced to prevent the transmission of transaction logs for changing data to other nodes. If users create a replication with the RECEIVE_ONLY option, the local server does not read the transaction logs of the remote server. Therefore, any replication issues such as network failures do not affect the local server system.
```

Author: If users create a replication with the RECEIVE_ONLY option -> If users create a replication with the RECEIVE_ONLY option on remote server, 가 맞을것 같네요

##### Meta Version

Given that the major version of Meta has changed, **it is necessary to reconfigure the metadata when upgrading from the earlier version to Altibase 7.3**.

##### Communication Protocol Version

The patch version of the communication protocol version has been changed. Altibase 7.3 ensures client backward compatibility only if the versions of major and minor are identical.

> Client backward compatibility ensures that user applications developed using an older version of the Altibase library work well when connected to a newer version of Altibase.

##### Replication Protocol Version

The replication protocol version indicates the compatibility of replication.

Altibase 7.3 guarantees replication backward compatibility in LAZY mode because there are no changes in the major and minor versions of the replication protocol. However, it does not guarantee compatibility with replication additional features including EAGER mode, because the patch version of the replication protocol has been changed.

> ###### Altibase Replication Backward Compatibility
>
> Only LAZY mode replication guarantees backward compatibility.
>

> > While EAGER mode replication, DDL synchronization and Offline Replication do not ensure backward compatibility since they require the same replication protocol version.

Author: ```suggestion
> EAGER mode replication, DDL synchronization and Offline Replication do not guarantee backward compatibility since they require the same replication protocol version.
```
while이 잘못들어간것 같네요.

###### CLIENT_TYPE

The CLIENT_TYPE of an Altibase 7.3 JDBC session is NEW_JDBC42.

</br>

##### Changes in SQL Results and Execution Plans

- Improved SQL Performance When Using the ORDER BY Clause in Subquery Inline Views

  This change has an impact on the execution plan of SQLs that utilize it, with the SORT plan node being eliminated within the SUBQUERY FILTER.

- Optimization of Nested LEFT OUTER JOIN Operations

  This change can result in alterations to the execution plan and potential differences in the SQL execution results for SQLs affected by it.

- Modifications and Additions to Subquery Unnesting Functionality

  This change can lead to alterations in the execution plans of SQLs influenced by it.

</br>

##### Replication Restrictions

###### Replication Restrictions Between Altibase 7.1 and Altibase 7.3

> DDL synchronization and Offline Replication are not supported between Altibase 7.1 and Altibase 7.3 when the running replication mode is EAGER.

Author: ```suggestion
DDL synchronization and Offline Replication are not supported between Altibase 7.1 and Altibase 7.3.
```
Eager replication 이 안되는건 일반적인 내용이라서 빼도 됩니다.
7.3에 추가된 이중화 제약사항이라서 ddl sync와 offline replication을 언급하면 될것같네요~

##### Meta Version

Given that the major version of Meta has changed, **it is necessary to reconfigure the metadata when upgrading from the earlier version to Altibase 7.3**.

##### Communication Protocol Version

The patch version of the communication protocol version has been changed. Altibase 7.3 ensures client backward compatibility only if the versions of major and minor are identical.

> Client backward compatibility ensures that user applications developed using an older version of the Altibase library work well when connected to a newer version of Altibase.

##### Replication Protocol Version

The replication protocol version indicates the compatibility of replication for both backward compatibility and compatibility with additional replication features.

Altibase replication in LAZY mode ensures the Altibase replication backward compatibility because there is no change in major and minor versions of the replication protocol. However, it does not ensure compatibility with replication additional features including EAGER mode replication, because the patch version has been changed.

> 

Author: Altibase replication 이 중복해서 나오길래 좀 봤는데,, 
1. The replication protocol version indicates the compatibility of replication for both backward compatibility and compatibility with additional replication features. 
-> The replication protocol version indicates the compatibility of replication.

2. Altibase 7.3 ensures replication backward compatibility in LAZY mode but does not ensure it in replication additional features including EAGER mode replication. Because the patch version has been changed, but there are no changes in the major and minor versions of the replication protocol. 이게 더 나은것 같은데요??

Author: 그럼 이렇게 해주세요~
The replication protocol version indicates the compatibility of replication.
Altibase 7.3 guarantees replication backward compatibility in LAZY mode because there are no changes in the major and minor versions of the replication protocol. However, it does not guarantee compatibility with replication additional features including EAGER mode, because the patch version of the replication protocol has been changed.

Reviewer: 수정하고 로컬에서 저장했습니다. 수정 사항이 더 쌓이면 후에 커밋하겠습니다.

> ###### Altibase Replication Backward Compatibility
>
> Only LAZY mode replication guarantees backward compatibility.
>

> > While EAGER mode replication DDL synchronization and Offline Replication do not ensure backward compatibility since they require the same replication protocol version.

Author: ```suggestion
> While EAGER mode replication, DDL synchronization and Offline Replication do not ensure backward compatibility since they require the same replication protocol version.
```

##### Simple Query Optimization on Memory Partitioned Tables

Previously, simple query optimization was only supported for memory tables. But now it extendes support to memory partitioned tables and enhanced DML performance of memory partitioned tables.

##### SERIAL FILTER EXEUTE

Improved the performance of performing row filters by serializing the Filter operator and optimizing the function call structure. The SERIAL_FILTER hint and SERIAL_EXECUTE_MODE properties have been added to enable this feature. You can see FILTER SERIAL EXECUTE in the execution plan. Serial Filter Enable can be confirmed by checking for 'FILTER SERIAL EXECUTE' in execution plan.

##### Performance Enhancement for Scalar Subqueries

Optimized the execution of scalar subqueries to improve their performance.

##### Performance Enhancement in the FOR Loop Clause of PSM

##### Performance Enhancement in Replication Sender

- Added functionality to decompress only the logs required for replication in compressed logs.
- Changed the xLog compression algorithm from LZO to LZ4.

##### Performance Enhancement for Migration

> Improved performance of data insertion for large data migrations. A new option, -lightmod, has been introduced of iloader. Refer to [**iLoader User's Maunal**](https://github.com/ALTIBASE/Documents/blob/master/Manuals/Altibase_trunk/eng/iLoader%20User's%20Manual.md).

Reviewer: - 오타: -lightmode 오타가 있습니다(국문에서도 발견됐습니다)
